solr/core/src/java/org/apache/solr/cloud/ElectionContext.java

```diff
@@ -63,7 +64,7 @@
   final ZkNodeProps leaderProps;
   final String id;
   final String leaderPath;
-  volatile String leaderSeqPath;
+  final AtomicReference<String> leaderSeqPath = new AtomicReference<>();
```

Comment:
I found a few instances where the AtomicRef is used (directly) in a log message, but there is a local `path` variable that would have the correct value in it.